### ISSUE TITLE
Obviate Rollup

### ISSUE THREAD

**Issue by kriskowal** [Agoric/Agoric#9686]
## What is the Problem Being Solved?

Rollup takes an entry point ESM and converts it into a single script. This operation erases the boundaries between packages so an auditor cannot reason about lexical accessibility of powerful objects. We currently use Rollup for Core Eval proposals and bootstrapping scripts in the kernel. Contracts use a Zip archive bundle format, but we have to bootstrap the runtime for that by evaluating a script.

## Description of the Design

Long ago, we used Rollup to bundle SES. However, the SES emulation of ESM includes a mechanism for converting individual modules into scripts, and employs a calling convention to link them. In a feat of meta-circularity, we are now a bootstrap SES script by using its ESM emulator to generate a single script with inline linkage. MetaMask grew that bundler out to support CJS. It’s a hair’s breadth away from supporting JSON modules too. It also supports all the Endo compartment mapper support for Node.js “build tags” or “predicates” in `package.json` `"exports"`. We also already have a bundler CLI, `bundle-source`.

We can:

1. Generalize `bundle-source` to support other bundle formats including `endoZipBase64` (default), and our bootstrapping formats `nestedEvalute` and `getExport`.
2. Add a `endoScript` format that uses the Endo bundler to generate a script.
3. Add a no-cache variant so usage is as familiar as Rollup.

In the end, we would replace Rollup commands with:

```sh
bundle-source --format endoScript entry.js | jq .source -r > bundle.js
```

We would continue to also support a cache mode:

```sh
bundle-source -f endoScript --cache-js bundle/ proposal.js proposal
```

## Security Considerations

Increases ceremony in the reviewable core eval scripts in order to make lexical access to powers within individual modules more evident.

## Scaling Considerations

None.

## Test Plan

Manual.

## Upgrade Considerations

None.

## Estimate

Should take an afternoon to implement.

## Implications

With this implemented,

- We can obviate Rollup for Core Eval and improve the audit experience
- `bundle-source` becomes a one-stop shop for all bundling needs for proposing contracts
- We dogfood more of Endo
- We can investigate using this format instead of `nestedEvaluate` for bootstrapping SwingSet workers
- With an additional `--tag` flag so `--tag=xs` can be specified, we make it easier to generate endo test262 scripts for integration testing in https://github.com/endojs/endo/issues/400

**Comment by kriskowal** [comment]
This is integrated.
